### PR TITLE
Explosive candles

### PR DESCRIPTION
**Pull request by heepox**
## About The Pull Request

Candles that go boom when they run out of wax (need to adjust explosion maybe and wax amount on them)

## Why It's Good For The Game

more trickery and a new traitor item for mime for 3tc a box of 5 candles but they explode after a long time

## Changelog
:cl:
add: Explosive candles
/:cl:

### CODE REVIEW

**Comment by heepox** [comment]
oh and also coded in 10 minutes so maybe some spaghetti there?

File: code/game/objects/items/candle.dm [hunk 1]
```diff
@@ -63,7 +64,10 @@
 		return PROCESS_KILL
 	if(!infinite)
 		wax--
-	if(!wax)
+	if(explosive && !wax)
+		explosion(src.loc,0,2,3,flame_range = 3)
+		qdel(src)
+	if(!wax && !explosive)
 		new /obj/item/trash/candle(loc)
 		qdel(src)
```
Comment (utilbird) [inline code/game/objects/items/candle.dm]:
```suggestion
	if(!wax)
		if(explosive)
			explosion(src.loc,0,2,3,flame_range = 3)
		else
			new /obj/item/trash/candle(loc)
		qdel(src)
```

Comment (heepox) [reply code/game/objects/items/candle.dm]: tested and works just fine like this. thanks.

**Comment by eeSPee** [comment]
Doesn't seem worth 3 TC to me. Or even 1, to be honestly.

Possibly make them craftable in game or contraband?

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -2106,6 +2106,15 @@ datum/uplink_item/role_restricted/superior_honkrender
 	restricted_roles = list("Mime")
 	surplus = 0
 
+/datum/uplink_item/role_restricted/explosive_candles
+	name = "Trick Candles"
+	desc = "Explosive candles that go boom when they finish burning. A very unexpected explosive with a long fuse. \
```
Comment (Archanial) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
	desc = "Explosive candles that explode when they finish burning. A very unexpected explosive with a long fuse. \
```

File: code/game/objects/items/candle.dm [hunk 3]
```diff
@@ -64,7 +65,10 @@
 	if(!infinite)
 		wax--
 	if(!wax)
-		new /obj/item/trash/candle(loc)
+		if(explosive)
+			explosion(src.loc,0,2,3,flame_range = 3)
```
Comment (Archanial) [inline code/game/objects/items/candle.dm:69]:
```suggestion
			explosion(loc, 0, 2, 3, flame_range = 3)
```

**Review by Archanial — CHANGES_REQUESTED** [review]
They are not worth it, they should be craftable (maybe after injecting candle with plasma?)
Also, making it mime only makes no sense.

**Comment by ike709** [comment]
Should just be the result of injecting plasma into a candle, not a traitor item. Otherwise a decent idea.

**Comment by heepox** [comment]
> Should just be the result of injecting plasma into a candle, not a traitor item. Otherwise a decent idea.

right ill do that later.

**Review by PowerfulBacon — APPROVED** [review]
Seems alright, but since the entire candle has to burn out before it explodes I feel like it will have limited use in game

**Comment by FortSumter** [comment]
Plasma candles that explode when lit would be funny; this doesn't need a timer nor to be an mime exclusive tot item. The main problems is that I see absolutely no use for these at all since only the chaplain uses candles regularly.

**Comment by ike709** [comment]
It should definitely explode when lit not when it burns out.

**Comment by ike709** [comment]
🔑  because stale

**Comment by francinum** [comment]
🔑

**Pull request closed without merging**